tests/YasumiBase.php

```diff
@@ -287,6 +287,111 @@ public function generateRandomEasterMondayDates($timezone = 'UTC', $iterations =
+     * holiday day is moved to to Monday.
+     *
+     * @param int    $month      month (number) for which the test date needs to be generated
+     * @param int    $day        day (number) for which the test date needs to be generated
+     * @param string $timezone   name of the timezone for which the dates need to be generated
+     * @param int    $iterations number of iterations (i.e. samples) that need to be generated (default: 10)
+     * @param int    $range      year range from which dates will be generated (default: 1000)
+     *
+     * @return array list of random test dates used for assertion of holidays.
+     */
+    public function generateRandomDatesWithHolidayMovedToMonday(
+        $month,
+        $day,
+        $timezone = 'UTC',
+        $iterations = 10,
+        $range = 1000
+    ) {
+        return $this->generateRandomDatesWithModifier(
+            $month,
+            $day,
+            function ($year, \DateTime $date) {
+                if ($this->isWeekend($date)) {
+                    $date->modify('next monday');
+                }
+            },
+            $timezone,
+            $iterations,
+            $range
+        );
+    }
+
+    /**
+     * Returns a list of random test dates used for assertion of holidays with applied callback.
+     *
+     * @param int      $month      month (number) for which the test date needs to be generated
+     * @param int      $day        day (number) for which the test date needs to be generated
+     * @param callable $callback   callback(int $year, \DateTime $dateTime) to modify $dateTime by custom rules
+     * @param string   $timezone   name of the timezone for which the dates need to be generated
+     * @param int      $iterations number of iterations (i.e. samples) that need to be generated (default: 10)
+     * @param int      $range      year range from which dates will be generated (default: 1000)
+     *
+     * @return array list of random test dates used for assertion of holidays with applied callback.
+     */
+    public function generateRandomDatesWithModifier(
+        $month,
+        $day,
+        callable $callback,
+        $timezone = 'UTC',
+        $iterations,
+        $range
+    ) {
+        $data  = [];
+
+        for ($i = 1; $i <= $iterations; ++$i) {
+            $year = $this->generateRandomYear($range);
+            $date = new \DateTime("{$year}-{$month}-{$day}", new \DateTimeZone($timezone));
+
+            $callback($year, $date);
+
+            $data[] = [$year, $date->format('Y-m-d')];
+        }
+
+        return $data;
+    }
+
+    /**
+     * Checks if given $dateTime is a weekend.
+     *
+     * @param \DateTimeInterface $dateTime date for which weekend will be checked.
+     *
+     * @return bool true if $dateTime is a weekend, false otherwise.
+     */
+    public function isWeekend(\DateTimeInterface $dateTime)
+    {
+        return in_array((int) $dateTime->format('w'), [0, 6], true);
```

Comment:
Although in many countries weekend is considered to be Saturday and Sunday, in other countries it isn't. Need to be aware of that when using this function.